parsers/FR_O.py

```diff
@@ -157,92 +169,89 @@ def fetch_data(
         if data.get("errorcode") == "10002":
             raise ParserException(
                 "FR_O.py",
-                f"Rate limit exceeded. Please try again later after: {data.reset_time}",
+                f"Rate limit exceeded. Please try again later after: {data.get('reset_time')}",
             )
         elif data.get("error_code") == "ODSQLError":
             raise ParserException(
                 "FR_O.py",
                 "Query malformed. Please check the parameters. If this was previously working there has likely been a change in the API.",
             )
-    return data, DATE_STRING_MAPPING[zone_key]
+    if isinstance(data, list):
```

Comment:
One suggestion I would make here is to invert the condition so that you raise first if there an issue, otherwise you continue your flow path.
It limits the depth of the function and improves readability in my opinion :)
```suggestion
    if not isinstance(data, list):
```